azurerm_storage_management_policy (2.66 crash)

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

terraform 1.0.1
azurerm 2.66.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

resource `azurerm_storage_management_policy`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```
resource "azurerm_storage_account" "storage_account" {
    name			= var.environement.variables.envSpecific.storageAccountName
    resource_group_name		= var.environement.variables.envSpecific.rgName
    location			= var.environement.variables.envSpecific.location
    account_tier		= "Standard"
    account_replication_type	= "LRS"
    account_kind                = "StorageV2"
    is_hns_enabled              = "true"
    large_file_share_enabled	= "false"
    min_tls_version		= "TLS1_2"
    access_tier			= "Hot"
    allow_blob_public_access	= "true"
    enable_https_traffic_only	= "true"
    network_rules {
        default_action		= "Deny"
    }
}

resource "azurerm_storage_management_policy" "storage_management_policy" {
    storage_account_id		= azurerm_storage_account.storage_account.id
    rule {
        name			= "BackupPolicy"
        enabled			= true
        filters {
            blob_types		= ["blockBlob"]
        }
        actions {
            base_blob {
                tier_to_cool_after_days_since_modification_greater_than = 1
            }
        }
    }
}
```

### Panic Output

```
11:18:11  module.hiddenname.module.storageaccount.azurerm_storage_management_policy.storage_management_policy: Creating...
11:18:12  ╷
11:18:12  │ Error: Plugin did not respond
11:18:12  │ 
11:18:12  │   with module.hiddenname.module.storageaccount.azurerm_storage_management_policy.storage_management_policy,
11:18:12  │   on modules/StorageAccount/storageaccount.tf line 24, in resource "azurerm_storage_management_policy" "storage_management_policy":
11:18:12  │   24: resource "azurerm_storage_management_policy" "storage_management_policy" {
11:18:12  │ 
11:18:12  │ The plugin encountered an error, and failed to respond to the
11:18:12  │ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
11:18:12  │ contain more details.
11:18:12  ╵
11:18:13  
11:18:13  Stack trace from the terraform-provider-azurerm_v2.66.0_x5 plugin:
11:18:13  
11:18:13  panic: reflect: call of reflect.Value.IsNil on array Value
11:18:13  
11:18:13  goroutine 246 [running]:
11:18:13  reflect.Value.IsNil(...)
11:18:13  	/opt/goenv/versions/1.16.3/src/reflect/value.go:1086
11:18:13  github.com/mitchellh/mapstructure.(*Decoder).decodeSlice(0xc00028c390, 0x0, 0x0, 0x6111fc0, 0xc0019f82d0, 0x606d0c0, 0xc0021c4420, 0x197, 0x8, 0x8)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:1086 +0x10ad
11:18:13  github.com/mitchellh/mapstructure.(*Decoder).decode(0xc00028c390, 0x0, 0x0, 0x6111fc0, 0xc0019f82d0, 0x606d0c0, 0xc0021c4420, 0x197, 0xc0021c4420, 0x197)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:460 +0x836
11:18:13  github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc00028c390, 0x6111fc0, 0xc0019f82d0, 0x0, 0xc0021c4420)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:389 +0xf0
11:18:13  github.com/mitchellh/mapstructure.Decode(0x6111fc0, 0xc0019f82d0, 0x5fde3e0, 0xc0021c4420, 0x1, 0xc0019b61b0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:297 +0xa5
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList(0xc00147d650, 0xc001f286c0, 0x3, 0x3, 0x6111fc0, 0xc0019f82d0, 0xc001097400, 0xc0015145c8, 0xc001f28690)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:127 +0xd2
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc00147d650, 0xc001f286c0, 0x3, 0x3, 0x6111fc0, 0xc0019f82d0, 0x0, 0x0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:103 +0xde
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setObject(0xc00147d650, 0xc001ca6100, 0x2, 0x2, 0x623b880, 0xc001f285d0, 0xc002574b40, 0x10, 0x60df260)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:225 +0x21a
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc00147d650, 0xc001ca6100, 0x2, 0x2, 0x623b880, 0xc001f285d0, 0x4, 0x16f8615)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:109 +0x1de
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList.func1(...)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:123
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList(0xc00147d650, 0xc0019f8310, 0x1, 0x1, 0x606d0c0, 0xc0021c4390, 0xc00109c640, 0x6a02c75, 0xc002574500)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:143 +0x29e
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc00147d650, 0xc0019f8310, 0x1, 0x1, 0x606d0c0, 0xc0021c4390, 0x1, 0x1)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:103 +0xde
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).WriteField(0xc00147d650, 0xc0019f8310, 0x1, 0x1, 0x606d0c0, 0xc0021c4390, 0x0, 0x0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:89 +0x3f9
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc000e66c80, 0x6a02c75, 0x4, 0x606d0c0, 0xc0021c4390, 0x0, 0x0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:227 +0x1df
11:18:13  github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage.resourceStorageManagementPolicyRead(0xc000e66c80, 0x620a540, 0xc000ce7880, 0x0, 0x0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/storage_management_policy_resource.go:295 +0x525
11:18:13  github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage.resourceStorageManagementPolicyCreateOrUpdate(0xc000e66c80, 0x620a540, 0xc000ce7880, 0x0, 0x0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/storage_management_policy_resource.go:266 +0x8d0
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0010987e0, 0x715ab68, 0xc000fa9400, 0xc000e66c80, 0x620a540, 0xc000ce7880, 0x0, 0x0, 0x0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:318 +0x1ee
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0010987e0, 0x715ab68, 0xc000fa9400, 0xc0016cc700, 0xc0011a4780, 0x620a540, 0xc000ce7880, 0x0, 0x0, 0x0, ...)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:456 +0x67b
11:18:13  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000115068, 0x715ab68, 0xc000fa9400, 0xc0018acc80, 0xc000fa9400, 0x66d3400, 0xc0016c2700)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:955 +0x8ef
11:18:13  github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0018fbf00, 0x715ac10, 0xc000fa9400, 0xc0016cc5b0, 0xc0018fbf00, 0xc0016c2720, 0xc000af3ba0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:332 +0xb5
11:18:13  github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x66d3400, 0xc0018fbf00, 0x715ac10, 0xc0016c2720, 0xc00147f5c0, 0x0, 0x715ac10, 0xc0016c2720, 0xc001a7ef00, 0x4b0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
11:18:13  google.golang.org/grpc.(*Server).processUnaryRPC(0xc000228c40, 0x71a4538, 0xc0005c2600, 0xc00174a480, 0xc001166d20, 0xa67dd00, 0x0, 0x0, 0x0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
11:18:13  google.golang.org/grpc.(*Server).handleStream(0xc000228c40, 0x71a4538, 0xc0005c2600, 0xc00174a480, 0x0)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
11:18:13  google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000628ad0, 0xc000228c40, 0x71a4538, 0xc0005c2600, 0xc00174a480)
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
11:18:13  created by google.golang.org/grpc.(*Server).serveStreams.func1
11:18:13  	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd
11:18:13  
11:18:13  Error: The terraform-provider-azurerm_v2.66.0_x5 plugin crashed!
11:18:13  
11:18:13  This is always indicative of a bug within the plugin. It would be immensely
11:18:13  helpful if you could report the crash with the plugin's maintainers so that it
11:18:13  can be fixed. The output above should help diagnose the issue.
```

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

When creating a new resource: terraform apply
Crashes as well when refreshing the state of an existing  resource.


PS: Sorry for the bad formatting and everything, never used github before.

Same for me. After upgrading from 2.63 -> 2.66. Here is the output

```
 Error: Plugin did not respond
│ 
│   with module.ops.azurerm_storage_management_policy.housekeeping,
│   on modules/ops/main.tf line 84, in resource "azurerm_storage_management_policy" "housekeeping":
│   84: resource "azurerm_storage_management_policy" "housekeeping" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-azurerm_v2.66.0_x5 plugin:

panic: reflect: call of reflect.Value.IsNil on array Value

goroutine 470 [running]:
reflect.Value.IsNil(...)
	/opt/goenv/versions/1.16.3/src/reflect/value.go:1086
github.com/mitchellh/mapstructure.(*Decoder).decodeSlice(0xc001b82e30, 0x0, 0x0, 0x5513a40, 0xc0014dcbc0, 0x546ed40, 0xc00153fd88, 0x197, 0x8, 0x8)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/mitchellh/mapstructure/mapstructure.go:1086 +0x10ad
...
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:123
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList(0xc00153fc68, 0xc0014dcc10, 0x1, 0x1, 0x546ed40, 0xc00153fcf8, 0xc001030640, 0x5e04135, 0xc002bcec80)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:143 +0x29e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc00153fc68, 0xc0014dcc10, 0x1, 0x1, 0x546ed40, 0xc00153fcf8, 0x1, 0x1)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:103 +0xde
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).WriteField(0xc00153fc68, 0xc0014dcc10, 0x1, 0x1, 0x546ed40, 0xc00153fcf8, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_writer_map.go:89 +0x3f9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc000f10880, 0x5e04135, 0x4, 0x546ed40, 0xc00153fcf8, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:227 +0x1df
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage.resourceStorageManagementPolicyRead(0xc000f10880, 0x560bc60, 0xc000d77880, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/storage_management_policy_resource.go:295 +0x525
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc001026e00, 0x655b5c8, 0xc001a456c0, 0xc000f10880, 0x560bc60, 0xc000d77880, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:335 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc001026e00, 0x655b5c8, 0xc001a456c0, 0xc001bb9110, 0x560bc60, 0xc000d77880, 0xc001b831f0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:624 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0010ba0a8, 0x655b5c8, 0xc001a456c0, 0xc001a45700, 0xc001a456c0, 0x40ba05, 0x591b140)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:575 +0x43b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc001bfc1c0, 0x655b670, 0xc001a456c0, 0xc000f1fda0, 0xc001bfc1c0, 0xc001bdfda0, 0xc0013aaba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x5ad47c0, 0xc001bfc1c0, 0x655b670, 0xc001bdfda0, 0xc000f1fd40, 0x0, 0x655b670, 0xc001bdfda0, 0xc0012cd400, 0x3f9)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005821c0, 0x65a4dd8, 0xc001bfed80, 0xc001bf2c60, 0xc000a20810, 0x9a7ab30, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0005821c0, 0x65a4dd8, 0xc001bfed80, 0xc001bf2c60, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000bfeec0, 0xc0005821c0, 0x65a4dd8, 0xc001bfed80, 0xc001bf2c60)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd

Error: The terraform-provider-azurerm_v2.66.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

##[error]Bash exited with code '1'.
```

--- a/azurerm/internal/services/storage/storage_management_policy_resource.go
+++ b/azurerm/internal/services/storage/storage_management_policy_resource.go
@@ -292,7 +292,7 @@ func resourceStorageManagementPolicyRead(d *pluginsdk.ResourceData, meta interfa
 	if policy := result.Policy; policy != nil {
 		policy := result.Policy
 		if rules := policy.Rules; rules != nil {
-			if err := d.Set("rule", flattenStorageManagementPolicyRules(policy.Rules)); err != nil {
+			if err := d.Set("rule", flattenStorageManagementPolicyRules(rules)); err != nil {
 				return fmt.Errorf("flattening `rule`: %+v", err)
 			}
 		}
@@ -490,7 +490,7 @@ func flattenStorageManagementPolicyRules(armRules *[]storage.ManagementPolicyRul
 
 				filter["match_blob_index_tag"] = flattenAzureRmStorageBlobIndexMatch(armFilter.BlobIndexMatch)
 
-				rule["filters"] = [1]interface{}{filter}
+				rule["filters"] = []interface{}{filter}
 			}
 
 			armAction := armDefinition.Actions
@@ -511,7 +511,7 @@ func flattenStorageManagementPolicyRules(armRules *[]storage.ManagementPolicyRul
 						intTemp := int(*armActionBaseBlob.Delete.DaysAfterModificationGreaterThan)
 						baseBlob["delete_after_days_since_modification_greater_than"] = intTemp
 					}
-					action["base_blob"] = [1]interface{}{baseBlob}
+					action["base_blob"] = []interface{}{baseBlob}
 				}
 
 				armActionSnaphost := armAction.Snapshot
@@ -526,7 +526,7 @@ func flattenStorageManagementPolicyRules(armRules *[]storage.ManagementPolicyRul
 					if armActionSnaphost.TierToCool != nil && armActionSnaphost.TierToCool.DaysAfterCreationGreaterThan != nil {
 						coolAfterCreation = int(*armActionSnaphost.TierToCool.DaysAfterCreationGreaterThan)
 					}
-					action["snapshot"] = [1]interface{}{map[string]interface{}{
+					action["snapshot"] = []interface{}{map[string]interface{}{
 						"delete_after_days_since_creation_greater_than":    deleteAfterCreation,
 						"change_tier_to_archive_after_days_since_creation": archiveAfterCreation,
 						"change_tier_to_cool_after_days_since_creation":    coolAfterCreation,
@@ -551,7 +551,7 @@ func flattenStorageManagementPolicyRules(armRules *[]storage.ManagementPolicyRul
 					}}
 				}
 
-				rule["actions"] = [1]interface{}{action}
+				rule["actions"] = []interface{}{action}
 			}
 		}
 

--- a/azurerm/internal/services/storage/storage_management_policy_resource.go
+++ b/azurerm/internal/services/storage/storage_management_policy_resource.go
@@ -544,7 +544,7 @@ func flattenStorageManagementPolicyRules(armRules *[]storage.ManagementPolicyRul
 					if armActionVersion.TierToCool != nil && armActionVersion.TierToCool.DaysAfterCreationGreaterThan != nil {
 						coolAfterCreation = int(*armActionVersion.TierToCool.DaysAfterCreationGreaterThan)
 					}
-					action["version"] = [1]interface{}{map[string]interface{}{
+					action["version"] = []interface{}{map[string]interface{}{
 						"delete_after_days_since_creation":                 deleteAfterCreation,
 						"change_tier_to_archive_after_days_since_creation": archiveAfterCreation,
 						"change_tier_to_cool_after_days_since_creation":    coolAfterCreation,

--- a/azurerm/internal/services/storage/storage_management_policy_resource_test.go
+++ b/azurerm/internal/services/storage/storage_management_policy_resource_test.go
@@ -25,7 +25,7 @@ func TestAccStorageManagementPolicy_basic(t *testing.T) {
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 				check.That(data.ResourceName).Key("rule.#").HasValue("1"),
-				check.That(data.ResourceName).Key("rule.0.name").HasValue("rule1"),
+				check.That(data.ResourceName).Key("rule.0.name").HasValue("rule-1"),
 				check.That(data.ResourceName).Key("rule.0.enabled").HasValue("true"),
 				check.That(data.ResourceName).Key("rule.0.filters.#").HasValue("1"),
 				check.That(data.ResourceName).Key("rule.0.filters.0.prefix_match.#").HasValue("1"),

